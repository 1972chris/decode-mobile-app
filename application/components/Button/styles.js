import { StyleSheet } from 'react-native';

const button = {
  alignSelf: 'stretch',
  borderRadius: 2,
  elevation: 2,
  height: 36,
  marginBottom: 18,
  marginHorizontal: 16,
  marginTop: 10,
  paddingHorizontal: 16,
  paddingVertical: 8,
  shadowColor: 'rgba(0, 0, 0, 0.54)',
  shadowOffset: { width: 0, height: 2 },
  shadowOpacity: 0.8,
  shadowRadius: 2,
};

const buttonText = {
  alignSelf: 'center',
  fontSize: 16,

  fontWeight: '500',
};

const styles = StyleSheet.create({
  signButton: {
    ...button,
    backgroundColor: 'rgb(0,163,158)',
  },
  signButtonDisabled: {
    ...button,
    backgroundColor: 'rgba(0,163,158,0.4)',
  },
  buttonText: {
    ...buttonText,
    color: '#FFF',
  },
  disabledButtonText: {
    ...buttonText,
    color: 'rgba(255, 255, 255, 0.6)',
  },
});

export default styles;
