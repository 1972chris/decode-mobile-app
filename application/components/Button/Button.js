import React from 'react';
import { StyleSheet, Text, TouchableOpacity, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

function Button(props) {
  const buttonTextStyle = props.enabled ? styles.buttonText : styles.disabledButtonText;
  const buttonStyle = props.enabled ? styles.signButton : styles.signButtonDisabled;

  const style = StyleSheet.create(props.style);

  return (
    <TouchableOpacity
      disabled={!props.enabled}
      style={[buttonStyle, style]}
      onPress={props.onPress}
    >
      <Text style={buttonTextStyle}>{props.name}</Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  enabled: PropTypes.bool,
  onPress: PropTypes.func,
  name: PropTypes.string.isRequired,
  style: ViewPropTypes.style, // eslint-disable-line
};

Button.defaultProps = {
  onPress: () => {
  },
  enabled: true,
  style: {},
};

export default Button;

