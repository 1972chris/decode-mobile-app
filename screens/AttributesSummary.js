/*
 * DECODE App – A mobile app to control your personal data
 * Copyright (C) 2019 – Thoughtworks Ltd.
 * Copyright (C) 2019 – DRIBIA Data Research S.L.
 *
 * DECODE App is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * DECODE App is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * email: ula@dribia.com
 */

import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { View, Text, Dimensions, ScrollView, Platform } from 'react-native';
import { translate } from 'react-i18next';
import { goToAttributesVerification } from '../application/redux/actions/navigation';
import LinkButton from '../application/components/LinkButton/LinkButton';
import RequesterInfo from '../application/components/RequesterInfo/RequesterInfo';
import PetitionDescription from '../application/components/PetitionDescription/PetitionDescription';
import Logo from '../application/components/ScreenLogo/ScreenLogo';
import openPetitionInBrowser from '../application/utils';
import styles from './styles';
import i18n from '../i18n';

class AttributesSummary extends React.Component {
  static route = {
    navigationBar: {
      backgroundColor: 'white',
      borderBottomWidth: 0,
      borderBottomColor: 'white',
      elevation: 1,
      fontSize: 20,
      fontWeight: '500',
      tintColor: 'rgb(0,163,158)',
      renderTitle: () => <View  style={{marginLeft: Platform.OS === 'ios' ? -20 : -60,}} ><Logo/></View>,
      height: 80,
    },
  };

  render() {
    const { petition, t } = this.props;
    const {height: windowHeight} = Dimensions.get('window');
    return (
      <ScrollView style={styles.petitionSummaryContainer} contentContainerStyle={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between',
        minHeight: windowHeight,
      }}>
        <View style={{ flex: 1 }}>
          <PetitionDescription
            title={petition.title}
            description={petition.description}
          />

          <View
            style={{
              marginTop: 10,
              borderBottomColor: '#80979797',
              borderBottomWidth: 1,
            }}
          />

          <RequesterInfo name={petition.requester} />

          <View style={{ paddingBottom: 20 }}>
            <Text style={{ color: '#9B9B9B', fontSize: 14 }}>
              <Text style={{ color: '#D0021B' }}>*</Text> {t('requiredAttributes')}
            </Text>
          </View>

          <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'space-between' }}>
            {
              petition.attributes.mandatory[0].verificationInput ? (
                <React.Fragment>
                  <Text>
                    {t(petition.attributes.mandatory[0].predicate)} <Text style={{ color: '#D0021B' }}>*</Text>
                  </Text>
                  <LinkButton
                    name={t('button')}
                    onPress={() => this.props.goToAttributesVerification()}
                    style={{
                      marginTop: 40,
                      alignSelf: 'center',
                    }}
                  />
                </React.Fragment>
              ) : null
            }
          </View>

          <View style={{
            alignItems: 'center',
            justifyContent: 'flex-end',
            paddingBottom: 20,
          }}
          >
            <Text
              style={styles.cancelSigningPetition}
              onPress={() => openPetitionInBrowser(petition.id)}
            >{t('cancel')}
            </Text>
          </View>
        </View>

      </ScrollView>
    );
  }
}

AttributesSummary.propTypes = {
  petition: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    closingDate: PropTypes.string,
    attributes: PropTypes.shape({
      mandatory: PropTypes.arrayOf(PropTypes.shape({
        provenance: PropTypes.shape({
          url: PropTypes.string,
        }),
      })),
    }),
  }),
  goToAttributesVerification: PropTypes.func.isRequired,
  walletId: PropTypes.string.isRequired,
  t: PropTypes.func.isRequired,
};

AttributesSummary.defaultProps = {
  petition: undefined,
};

const mapStateToProps = state => ({
  petition: state.petition.petition,
  walletId: state.wallet.id,
});

const mapDispatchToProps = dispatch => ({
  goToAttributesVerification: () => { dispatch(goToAttributesVerification()); },
});

export default translate('attributesSummary', { i18n })(connect(mapStateToProps, mapDispatchToProps)(AttributesSummary));
