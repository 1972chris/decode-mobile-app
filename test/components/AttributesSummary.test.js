import React from 'react';
import { Text } from 'react-native';
import configureStore from 'redux-mock-store';
import fetchMock from 'fetch-mock';
import thunk from 'redux-thunk';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build/index';
import Button from '../../application/components/Button/Button';
import AttributesSummary from '../../screens/AttributesSummary';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

describe('AttributesSummary', () => {
  const somePetitionLink = 'http://some-petition.com';
  const goToPetitionSummaryMock = jest.fn();
  const initialState = {
    petitionLink: {
      petitionLink: 'http://something.com',
    },
    petition: {
      petitionAttributes: [],
      petition: {},
    },
    attributes: {
      list: new Map(),
    },
    wallet: {
      id: 'something',
    },
  };

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  describe('the page', () => {
    it('should show the verify button', () => {
      const store = mockStore(initialState);

      // WHEN I VISUALIZE THE Attribute Summary
      const wrapper = shallow(<AttributesSummary />)
        .first().shallow()
        .first()
        .shallow({ context: { store } });

      const buttonWrapper = wrapper.dive().find(Button);

      // Only one button
      expect(buttonWrapper).toHaveLength(1);
      expect(buttonWrapper.prop('name')).toEqual('Verificar');
      expect(buttonWrapper.first().prop('enabled')).toEqual(true);
    });

    it('should show the title of the petition', () => {
      const title = 'Petition Title';
      const state = {
        ...initialState,
        petition: {
          petition: {
            title,
            description: 'world',
            closingDate: 'today',
            id: '1234',
          },
          petitionAttributes: [],
        },
      };
      const store = mockStore(state);

      // WHEN I VISUALIZE THE Attribute Summary
      const wrapper = shallow(<AttributesSummary />)
        .first().shallow()
        .first()
        .shallow({ context: { store } });

      const TextWrappers = wrapper.dive().find(Text).findWhere(n => n.text() === title);
      expect(TextWrappers).toHaveLength(1);
    });

    describe('when the verify button is pressed', () => {
      xit('should go to credential issuer', () => {
        const wrapper = shallow(<AttributesSummary />)
          .first().shallow()
          .first()
          .shallow({ context: { store: mockStore(initialState) } });

        // wrapper.dive().find(Button).first().simulate('click');

        const spy = jest.spyOn(wrapper.instance(), 'openWebBrowserAsync');
        wrapper.update();
        wrapper.find(Button).simulate('click');
        expect(spy).toHaveBeenCalled();
      });
    });

    describe('on redirect from credential issuer', () => {
      xit('should go to petitionSummary page', () => {
        const wrapper = shallow(<AttributesSummary />)
          .first().shallow()
          .first()
          .shallow({ context: { store: mockStore(initialState) } });

        const attributesSummaryComponent = wrapper.dive().instance();
        attributesSummaryComponent.props = {
          ...attributesSummaryComponent.props,
          goToPetitionSummary: goToPetitionSummaryMock,
        };

        wrapper.dive().find(Button).first().simulate('click');

        expect(goToPetitionSummaryMock).toBeCalledWith(somePetitionLink);
      });
    });
  });

  describe('handleRedirect', () => {
    it('should add new attribute with credential', async () => {
      const someUrl = 'blah.com';
      const someEvent = {
        url: someUrl,
      };
      const addCredentialMock = jest.fn();
      const someWalletId = '123';
      const residencyAttribute = {};
      const somePetition = {
        attributes: {
          mandatory: [residencyAttribute],
          optional: [],
        },
      };
      const wrapper = shallow(<AttributesSummary />)
        .first().shallow()
        .first()
        .shallow({ context: { store: mockStore(initialState) } });

      const attributesSummaryComponent = wrapper.dive().instance();
      attributesSummaryComponent.props = {
        ...attributesSummaryComponent.props,
        petition: somePetition,
        walletId: someWalletId,
        petitionLink: somePetitionLink,
        goToPetitionSummary: goToPetitionSummaryMock,
        addCredential: addCredentialMock,
      };

      await attributesSummaryComponent.handleRedirect(someEvent);

      expect(addCredentialMock).toBeCalledWith(residencyAttribute, someWalletId, someUrl);
      expect(goToPetitionSummaryMock).toBeCalledWith(somePetitionLink);
    });
  });
});
