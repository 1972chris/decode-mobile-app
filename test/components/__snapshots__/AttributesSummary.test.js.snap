// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AttributesSummary page when rendered should match this snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AttributesSummary
    addCredential={[Function]}
    attributes={
      Object {
        "isRequiredAttributeEnabled": false,
        "list": Array [],
        "optionalAttributesToggleStatus": Object {
          "age": false,
          "gender": false,
        },
      }
    }
    getPetition={[Function]}
    goToPetitionSummary={[Function]}
    petition={Object {}}
    petitionLink="http://something.com"
    walletId="something"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "backgroundColor": "#13A398",
              "flex": 1.2,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 100,
                "marginVertical": 20,
                "resizeMode": "contain",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "700",
              }
            }
          >
            Isabella Dominguez
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#13A398",
                "fontSize": 20,
                "margin": 16,
                "marginTop": 30,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#3E393C",
                "fontSize": 13,
                "fontWeight": "900",
                "marginLeft": 16,
              }
            }
          >
            DECIDIM needs to verify your Barcelona residency.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#3E393C",
                "fontSize": 13,
                "fontWeight": "100",
                "marginLeft": 16,
                "marginTop": 8,
              }
            }
          >
            You can do this with the Barcelona City Council
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Button
            enabled={true}
            name="Verify with Barcelona Council"
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 40,
                "width": 250,
              }
            }
          />
        </View>,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "color": "#00F",
                "fontSize": 14,
                "fontWeight": "500",
                "marginVertical": 20,
                "textDecorationLine": "underline",
              }
            }
          >
            Or, cancel signing this petition
          </Text>
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 100,
                  "marginVertical": 20,
                  "resizeMode": "contain",
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "700",
                }
              }
            >
              Isabella Dominguez
            </Text>,
          ],
          "style": Object {
            "backgroundColor": "#13A398",
            "flex": 1.2,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "alignSelf": "center",
                "height": 100,
                "marginVertical": 20,
                "resizeMode": "contain",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Isabella Dominguez",
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "700",
              },
            },
            "ref": null,
            "rendered": "Isabella Dominguez",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#13A398",
                  "fontSize": 20,
                  "margin": 16,
                  "marginTop": 30,
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "900",
                  "marginLeft": 16,
                }
              }
            >
              DECIDIM needs to verify your Barcelona residency.
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "100",
                  "marginLeft": 16,
                  "marginTop": 8,
                }
              }
            >
              You can do this with the Barcelona City Council
            </Text>,
          ],
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": undefined,
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#13A398",
                "fontSize": 20,
                "margin": 16,
                "marginTop": 30,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "DECIDIM needs to verify your Barcelona residency.",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#3E393C",
                "fontSize": 13,
                "fontWeight": "900",
                "marginLeft": 16,
              },
            },
            "ref": null,
            "rendered": "DECIDIM needs to verify your Barcelona residency.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "You can do this with the Barcelona City Council",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#3E393C",
                "fontSize": 13,
                "fontWeight": "100",
                "marginLeft": 16,
                "marginTop": 8,
              },
            },
            "ref": null,
            "rendered": "You can do this with the Barcelona City Council",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            enabled={true}
            name="Verify with Barcelona Council"
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 40,
                "width": 250,
              }
            }
          />,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "enabled": true,
            "name": "Verify with Barcelona Council",
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "marginTop": 40,
              "width": 250,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "color": "#00F",
                "fontSize": 14,
                "fontWeight": "500",
                "marginVertical": 20,
                "textDecorationLine": "underline",
              }
            }
          >
            Or, cancel signing this petition
          </Text>,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Or, cancel signing this petition",
            "ellipsizeMode": "tail",
            "onPress": [Function],
            "style": Object {
              "alignSelf": "center",
              "color": "#00F",
              "fontSize": 14,
              "fontWeight": "500",
              "marginVertical": 20,
              "textDecorationLine": "underline",
            },
          },
          "ref": null,
          "rendered": "Or, cancel signing this petition",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "backgroundColor": "#13A398",
                "flex": 1.2,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 100,
                  "marginVertical": 20,
                  "resizeMode": "contain",
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "700",
                }
              }
            >
              Isabella Dominguez
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#13A398",
                  "fontSize": 20,
                  "margin": 16,
                  "marginTop": 30,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "900",
                  "marginLeft": 16,
                }
              }
            >
              DECIDIM needs to verify your Barcelona residency.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "100",
                  "marginLeft": 16,
                  "marginTop": 8,
                }
              }
            >
              You can do this with the Barcelona City Council
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Button
              enabled={true}
              name="Verify with Barcelona Council"
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "marginTop": 40,
                  "width": 250,
                }
              }
            />
          </View>,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#00F",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "marginVertical": 20,
                  "textDecorationLine": "underline",
                }
              }
            >
              Or, cancel signing this petition
            </Text>
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Image
                source={1}
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 100,
                    "marginVertical": 20,
                    "resizeMode": "contain",
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "700",
                  }
                }
              >
                Isabella Dominguez
              </Text>,
            ],
            "style": Object {
              "backgroundColor": "#13A398",
              "flex": 1.2,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {
                  "alignSelf": "center",
                  "height": 100,
                  "marginVertical": 20,
                  "resizeMode": "contain",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Isabella Dominguez",
                "ellipsizeMode": "tail",
                "style": Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "700",
                },
              },
              "ref": null,
              "rendered": "Isabella Dominguez",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#13A398",
                    "fontSize": 20,
                    "margin": 16,
                    "marginTop": 30,
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#3E393C",
                    "fontSize": 13,
                    "fontWeight": "900",
                    "marginLeft": 16,
                  }
                }
              >
                DECIDIM needs to verify your Barcelona residency.
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#3E393C",
                    "fontSize": 13,
                    "fontWeight": "100",
                    "marginLeft": 16,
                    "marginTop": 8,
                  }
                }
              >
                You can do this with the Barcelona City Council
              </Text>,
            ],
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": undefined,
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#13A398",
                  "fontSize": 20,
                  "margin": 16,
                  "marginTop": 30,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "DECIDIM needs to verify your Barcelona residency.",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "900",
                  "marginLeft": 16,
                },
              },
              "ref": null,
              "rendered": "DECIDIM needs to verify your Barcelona residency.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "You can do this with the Barcelona City Council",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#3E393C",
                  "fontSize": 13,
                  "fontWeight": "100",
                  "marginLeft": 16,
                  "marginTop": 8,
                },
              },
              "ref": null,
              "rendered": "You can do this with the Barcelona City Council",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              enabled={true}
              name="Verify with Barcelona Council"
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "marginTop": 40,
                  "width": 250,
                }
              }
            />,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "enabled": true,
              "name": "Verify with Barcelona Council",
              "onPress": [Function],
              "style": Object {
                "alignSelf": "center",
                "marginTop": 40,
                "width": 250,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              onPress={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#00F",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "marginVertical": 20,
                  "textDecorationLine": "underline",
                }
              }
            >
              Or, cancel signing this petition
            </Text>,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Or, cancel signing this petition",
              "ellipsizeMode": "tail",
              "onPress": [Function],
              "style": Object {
                "alignSelf": "center",
                "color": "#00F",
                "fontSize": 14,
                "fontWeight": "500",
                "marginVertical": 20,
                "textDecorationLine": "underline",
              },
            },
            "ref": null,
            "rendered": "Or, cancel signing this petition",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
